pycloudlib/lxd/instance.py

```diff
@@ -333,41 +333,31 @@ def wait_for_stop(self):
             time.sleep(1)
         raise TimeoutError
 
-    def _wait_for_cloudinit(self, *, raise_on_failure: bool):
+    def _wait_for_cloudinit(self):
         """Wait until cloud-init has finished.
 
         If the instance is a virtual machine, we need to wait for the
         lxd-agent to be ready before getting the cloud-init status.
-        To do that, we retry on vm instances if raise_on_failure
-        is specified.
-
-        :param raise_on_failure:
-            When `True`, if the process waiting for cloud-init exits non-zero
-            then this method will raise an `OSError`.
+        To do that, we retry on vm instances if error is raised.
         """
         if self.is_vm:
-            retries = [30, 45, 60, 75, 90, 105]
-
-            for sleep_time in retries:
+            # This isn't a hard timeout. We'll only timeout if
+            # _wait_for_cloudinit() throws exceptions
+            timeout = 300
+            start_time = time.time()
+            last_error = None
+            while time.time() < start_time + timeout:
                 try:
-                    super()._wait_for_cloudinit(
-                        raise_on_failure=True)
+                    super()._wait_for_cloudinit()
```

Comment:
I don't think this code will work anymore, right ? Since `_wait_for_cloudinit` will not raise the `OSError` anymore, we will never  properly wait for cloud-init status to be over.

Also, I am little bit worried about this change, because even though we can adapt the uaclient code to have that explicit behavior of checking the cloud-init status return, any user that will need to launch a VM will need to know beforehand that they should wait for the `lxd-agent` to be ready. This means they will need to replicate this for loop logic on their code, which I don't think it is optimal.

Reply:
I thought the OSError was coming from LXD itself before we called `cloud-init status`, which means we would continue to get the error even after this change. Is that not true? I tried testing locally, but I'm actually not seeing any of these errors anymore for LXD VMs? Am I just lucky or did these issues maybe get resolved and it's all a moot point anyway :smile: ?

Reply:
Those are good questions. Let me reproduce that change and run some ua tests so I can test what happens in that scenario